docs/pluggable-packages.md

```diff
@@ -0,0 +1,66 @@
+# Pluggable packaging
+
+## Background
+
+`idpbuilder` is a tool that aims to:
+- Allow developers to stand up a Kubernetes cluster with components that make up a Internal Developer Platform (IDP). 
+- Allow for tests to run against IDPs in Continuous Integration systems.
+- Standup a working IDP for demo purposes.
+
+It also aims to achieve the above goals while having a single dependency, Docker, at run time. 
+
+When implementing IDPs using open source projects, there is no one set of projects that fit the needs of all organizations because:
+1. No organization went through the same path to reach the point of needing a IDP.
+1. Past technology choices were made for specific needs of each organization. This results in organizationally unique technology inertia that may not be correctable in the near future.
+1. No industry consensus and standards for choosing specific IDP components.
+
+To fit the needs from different organizations, idpbuilder needs to be flexible in the what and how it can deploy different packages. Currently idpbuilder uses ArgoCD to install a [set of application](https://github.com/cnoe-io/idpbuilder/blob/56089e4ae3b27cf90641bfbff2a96c36dd5263e1/pkg/apps/resources.go#L20-L32), and they cannot be changed without modifying the source code.
+
+## Goals
+
+The proposal in this document should:
+
+1. Make the packages installed by `idpbuilder` configurable.
+1. Minimize the number of runtime dependencies necessary.
+1. Make it easy for end users to configure their packages.
+1. Support custom setup scripts and workflows that need to be executed in a defined order before a package becomes ready.
+
+## Proposal
+
+This document proposes the following:
+- Make ArgoCD and Argo Workflows hard requirements. 
+- Define packages as Argo CD Applications.
+- Imperative pipelines for configuring packages are handled with ArgoCD resource hooks and Argo Workflows.
+
+Currently, ArgoCD is a base requirement for both the AWS reference implementation and idpbuilder. 
```

Comment:
The fact that we used Argo Workflows for the Data IDP will also make it fit nicely here.

Reply:
We also need to provide details on where the Argo CD applications will come from. are we expecting the Argo CD apps to be pulled from remote git repositories or to utilize the embedded file system and git repository?

- if the latter, then adding support for remote repos will have to be discussed as a future plan.
- if the former, then we should discuss accessing the possibility to access git repositories as a dependency (with implications it might have on air-gapped environments etc)